File: JS/二叉树/二叉树的属性/100_相同的树.js
function TreeNode(val, left, right) {
    this.val = (val===undefined ? 0 : val)
    this.left = (left===undefined ? null : left)
    this.right = (right===undefined ? null : right)
}
// 构建二叉树的辅助函数
function buildTree(arr) {
    if (arr.length === 0) return null;
    const nodes = arr.map(val => val!== null? new TreeNode(val) : null);
    let i = 0, j = 1;
    while (j < arr.length) {
        if (nodes[i]!== null) {
            nodes[i].left = nodes[j];
            j++;
            if (j < arr.length) {
                nodes[i].right = nodes[j];
                j++;
            }
        }
        i++;
    }
    return nodes[0];
}

var isSameTree = function(p, q) {
    let flag = true
    const dfs = (p, q) => {
        if(!flag || (!p && !q)) return
        if((!p && q) || (p && !q) || p.val!==q.val){
            flag = false
            return
        }else{
            dfs(p.left, q.left)
            dfs(p.right, q.right)
        }
    }
    dfs(p, q)
    return flag
};

let P = [1,2,3], Q=[1,2,3]
const p = buildTree(P), q = buildTree(Q)
const result = isSameTree(p, q);
console.log(result);

File: Python/面试经典150题/二叉树/100_相同的树.py
from typing import Optional
from collections import deque
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def build_tree(level_order):
    """
    根据层序遍历数组构建二叉树
    :param level_order: List[int or None], 例如 [3,9,20,null,null,15,7]
    :return: TreeNode 或 None
    """
    if not level_order or level_order[0] is None:
        return None

    root = TreeNode(level_order[0])
    queue = deque([root])
    index = 1

    while queue and index < len(level_order):
        node = queue.popleft()

        if index < len(level_order) and level_order[index] is not None:
            node.left = TreeNode(level_order[index])
            queue.append(node.left)
        index += 1

        if index < len(level_order) and level_order[index] is not None:
            node.right = TreeNode(level_order[index])
            queue.append(node.right)
        index += 1

    return root

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if not p and not q:
            return True
        if not p or not q:
            return False
        if p.val != q.val:
            return False
        return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)





if __name__ == "__main__":
    solution = Solution()
    p = build_tree([1,2,3])
    q = build_tree([1,2,3])
    print(solution.isSameTree(p, q))